audb/core/info.py

```diff
@@ -184,14 +184,19 @@ def header(
         *,
         version: str = None,
         cache_root: str = None,
-) -> audformat.Database:
+        timeout: float = -1,
+) -> typing.Optional[audformat.Database]:
     r"""Load header of database.
 
     Args:
         name: name of database
         version: version of database
         cache_root: cache folder where databases are stored.
             If not set :meth:`audb.default_cache_root` is used
+        timeout: maximum wait time if another thread or process is already
+            accessing the database. If timeout is reached, ``None`` is
+            returned. If timeout < 0 the method will block until the
+            database can be accessed
```

Comment:
The problem is that this does not only affect `audb.info.header()`, but the header is loaded also in other `audb.info` functions.

As the `audb.info` functions should access the header that is stored outside of the flavor folders, I think we don't have to provide a `timeout` argument here and for the other functions (where it is missing at the moment), as it should never be blocked for a long time.

Reply:
Good point.

>  as it should never be blocked for a long time.

The problem is that we have no control how long it will take. If a process just started to download a large database right before a second process arrives that wants to read only the header, the second process has to wait until the first process is finished. To avoid this problem, I introduced a second lock file for the header only. The first process will only acquire to download the header, afterwards - while it continues downloading tables / media - the second process can already acquire the lock and read the header.

Reply:
I had another look and realized that `audb.info.header()` should not interfer with `audb.load()` as it stores the header under `<cache-root>/<name>/db.yaml` instead of `<cache-root>/<name>/<flavor-id>/db.yaml`.

So, I would propose the following changes:
1. remove the `add_audb_meta` argument from `load_header_wo_lock()` and handle it only inside `load_header()`
2. use `load_header_wo_lock()` in `audb.info` as I doubt that the header will cause trouble (maybe double check with experiment on a database with a large header. But even then I would handle it in another pull request if needed)
3. use a single `.lock` file as implemented before. 

The proposal of point 3. should even be feasible without going for 1. and 2. as the header files are different for `audb.load()` and `audb.info`.

Reply:
> use a single .lock file as implemented before.

Ok, so you suggest to not lock the header at all, right?

Reply:
Ah, now I get it. We revert the changes, but do not use a lock with `audb.info.header()` as it is anyway loading a different file.

Reply:
Ok, I renamed `load_header_wo_lock()` to `load_header()` so that loading the header does not require `.lock`.

Reply:
Looks good now.

Could you also please update the description of the pull request, so we don't think at a later stage we included the lock file for `load_header()` here. Instead you can add it to the points that we should consider before closing #108, very much like you propose for the dependency file.

Reply:
done